### PR TITLE
Configuration via plain environment variables

### PR DESCRIPTION
**Pull request by moio**
This simplifies configuration by eliminating the JSON file we had - now all configuration happens directly via environment variables.

### CODE REVIEW

**Review by admd — APPROVED** [review]
+1

File: README.md [hunk 1]
```diff
@@ -12,36 +12,29 @@ Servers are registered and can be managed from the Hub just like normal SUSE Man
  - one central SUSE Manager/Uyuni Server (hereinafter: "Hub")
  - two or more peripheral Servers registered to the Hub as Salt clients
  - any number of end clients registered to peripheral Servers
-
-You will need credentials to all Server XMLRPC APIs, including the Hub's.
+ - a host that can connect to all the peripheral Server's XMLRPC APIs (it can be the Hub itself)
+ - crdentials to all Server XMLRPC APIs, including the Hub's
```
Comment (chiaradiamarcelo) [inline README.md]:
```suggestion
 - credentials to all Server XMLRPC APIs, including the Hub's
```

File: README.md [hunk 2]
```diff
@@ -12,36 +12,29 @@ Servers are registered and can be managed from the Hub just like normal SUSE Man
...
-       "manager_api_url": "http://localhost/rpc/api"
-   },
-    "connect_timeout": 10,
-    "read_write_timeout": 10,
-   }
- ```
+**If you are installing `hub-xmlrpc-api` on the Hub** then no further configuration is necessary. The daemon can be started via systemd as a `hub-xmlrpc-api.service` file is provided by the package.
 
-Replace `localhost` in `http://localhost/rpc/api` above with the Hub's FQDN if necessary.
+**If you are installing `hub-xmlrpc-api` on a sepeate host** then you should make sure that the URL to the Hub XMLRPC API is correctly configured in the `/etc/hub/hub.conf` file.
```
Comment (chiaradiamarcelo) [inline README.md]:
```suggestion
**If you are installing `hub-xmlrpc-api` on a separate host** then you should make sure that the URL to the Hub XMLRPC API is correctly configured in the `/etc/hub/hub.conf` file.
```

File: README.md [hunk 3]
```diff
@@ -12,36 +12,29 @@ Servers are registered and can be managed from the Hub just like normal SUSE Man
...
+## Additional configuration
 
-## Usage
+`/etc/hub/hub.conf` contains the following configuration parameters:
+ - `HUB_API_URL`: URL to the Hub XMLRPC API endpoint
+ - `HUB_CONNECT_TIMEOUT`: maximum number of seconds to wait for a response when connecting to a Server
+ - `HUB_REQUEST_TIMEOUT`: maximum numbr of seconds to wait for a response when calling a Server method
 
-`hub-xmlrpc-api` is a daemon, currently to be launched from the command line.
+ Defaults values should suffice in most settings.
```
Comment (chiaradiamarcelo) [inline README.md]:
```suggestion
 Default values should suffice in most settings.
```

File: README.md [hunk 4]
```diff
@@ -12,36 +12,29 @@ Servers are registered and can be managed from the Hub just like normal SUSE Man
...
 
+## Usage
 
-Once running, you can connect to the `hub-xmlrpc-api` at port 2830 via any XMLRPC compliant client libraries (see examples below).
+Once the `hub-xmlrpc-api` service is running, you can connect to it at port 2830 via any XMLRPC compliant client libraries (see examples below).
```
Comment (chiaradiamarcelo) [inline README.md]:
```suggestion
Once the `hub-xmlrpc-api` service is running, you can connect to it at port 2830 via any XMLRPC compliant client library (see examples below).
```

File: config/config.go [hunk 5]
```diff
@@ -1,51 +1,33 @@
...
-
-type HubConfig struct {
-	SUMA_API_URL string
-}
 
+// Config contains configuration parameters for this program
 type Config struct {
-	Hub                              HubConfig
-	ConnectTimeout, ReadWriteTimeout int
+	HubAPIURL                      string
```
Comment (chiaradiamarcelo) [inline config/config.go:13]:
```suggestion
	HubAPIEndpoint                      string
```

File: config/config.go [hunk 6]
```diff
@@ -1,51 +1,33 @@
...
-		return &Config{
-			Hub: HubConfig{
-				SUMA_API_URL: k.String("hub.manager_api_url"),
-			},
-			ConnectTimeout:   k.Int("connect_timeout"),
-			ReadWriteTimeout: k.Int("read_write_timeout"),
-		}
+
+	k.Load(confmap.Provider(map[string]interface{}{
+		"HUB_API_URL":         "http://localhost/rpc/api",
```
Comment (chiaradiamarcelo) [inline config/config.go:21]:
```suggestion
		"HUB_API_ENDPOINT":         "http://localhost/rpc/api",
```

File: config/config.go [hunk 7]
```diff
@@ -1,51 +1,33 @@
...
+		"HUB_CONNECT_TIMEOUT": 10,
+		"HUB_REQUEST_TIMEOUT": 10,
+	}, "."), nil)
+
+	k.Load(env.Provider("HUB_", ".", nil), nil)
+
+	return &Config{
+		HubAPIURL:      k.String("HUB_API_URL"),
```
Comment (chiaradiamarcelo) [inline config/config.go:29]:
```suggestion
		HubAPIEndpoint:      k.String("HUB_API_ENDPOINT"),
```

File: hub_api_gateway.go [hunk 8]
```diff
@@ -41,12 +41,12 @@ func initServer() {
 	serverSessionRepository := session.NewInMemoryServerSessionRepository(&syncMap)
 
 	//init gateway
-	serverAuthenticator := gateway.NewServerAuthenticator(conf.Hub.SUMA_API_URL, uyuniAuthenticator, uyuniTopologyInfoRetriever, hubSessionRepository, serverSessionRepository)
-	hubLoginer := gateway.NewHubLoginer(conf.Hub.SUMA_API_URL, uyuniAuthenticator, serverAuthenticator, uyuniTopologyInfoRetriever, hubSessionRepository)
-	hubLogouter := gateway.NewHubLogouter(conf.Hub.SUMA_API_URL, uyuniAuthenticator, hubSessionRepository)
+	serverAuthenticator := gateway.NewServerAuthenticator(conf.HubAPIURL, uyuniAuthenticator, uyuniTopologyInfoRetriever, hubSessionRepository, serverSessionRepository)
```
Comment (chiaradiamarcelo) [inline hub_api_gateway.go:44]:
```suggestion
	serverAuthenticator := gateway.NewServerAuthenticator(conf.HubAPIEndpoint, uyuniAuthenticator, uyuniTopologyInfoRetriever, hubSessionRepository, serverSessionRepository)
```

File: hub_api_gateway.go [hunk 9]
```diff
@@ -41,12 +41,12 @@ func initServer() {
...
+	hubLoginer := gateway.NewHubLoginer(conf.HubAPIURL, uyuniAuthenticator, serverAuthenticator, uyuniTopologyInfoRetriever, hubSessionRepository)
```
Comment (chiaradiamarcelo) [inline hub_api_gateway.go:45]:
```suggestion
	hubLoginer := gateway.NewHubLoginer(conf.HubAPIEndpoint, uyuniAuthenticator, serverAuthenticator, uyuniTopologyInfoRetriever, hubSessionRepository)
```

File: hub_api_gateway.go [hunk 10]
```diff
@@ -41,12 +41,12 @@ func initServer() {
...
+	hubLogouter := gateway.NewHubLogouter(conf.HubAPIURL, uyuniAuthenticator, hubSessionRepository)
```
Comment (chiaradiamarcelo) [inline hub_api_gateway.go:46]:
```suggestion
	hubLogouter := gateway.NewHubLogouter(conf.HubAPIEndpoint, uyuniAuthenticator, hubSessionRepository)
```

File: hub_api_gateway.go [hunk 11]
```diff
@@ -41,12 +41,12 @@ func initServer() {
...
 
-	hubProxy := gateway.NewHubProxy(conf.Hub.SUMA_API_URL, uyuniCallExecutor)
-	hubTopologyInfoRetriever := gateway.NewTopologyInfoRetriever(conf.Hub.SUMA_API_URL, uyuniTopologyInfoRetriever)
+	hubProxy := gateway.NewHubProxy(conf.HubAPIURL, uyuniCallExecutor)
```
Comment (chiaradiamarcelo) [inline hub_api_gateway.go:48]:
```suggestion
	hubProxy := gateway.NewHubProxy(conf.HubAPIEndpoint, uyuniCallExecutor)
```

File: hub_api_gateway.go [hunk 12]
```diff
@@ -41,12 +41,12 @@ func initServer() {
...
+	hubTopologyInfoRetriever := gateway.NewTopologyInfoRetriever(conf.HubAPIURL, uyuniTopologyInfoRetriever)
```
Comment (chiaradiamarcelo) [inline hub_api_gateway.go:49]:
```suggestion
	hubTopologyInfoRetriever := gateway.NewTopologyInfoRetriever(conf.HubAPIEndpoint, uyuniTopologyInfoRetriever)
```

**Review by chiaradiamarcelo — APPROVED** [review]

**Comment by chiaradiamarcelo** [comment]
Nice @moio !

**Pull request merged**